flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -497,6 +497,12 @@ public int unsynchronizedGetNumberOfQueuedBuffers() {
         return Math.max(buffers.size(), 0);
     }
 
+    @Override
+    public long unsynchronizedGetSizeOfQueuedBuffers() {
+        // Pretty rough approximation of real queue size.
+        return Math.max(unsynchronizedGetNumberOfQueuedBuffers() * bufferSize, 0);
```

Comment:
This is pretty bad since the queue can hold events that have different sizes, buffers of old buffer size, and unfinished buffers. But it is the easiest way to calculate queue size without dramatically rewriting code.

Reply:
I have a feeling that in this implementation this metric doesn't really tell us anything more compared to output queue length and desired buffer size. Probably we should provide a more accurate estimate/metric or just drop it.

The question would be what to do with this for the input bytes length. As this is only for local input channels, maybe that's a good enough estimation?

Reply:
I think for only local channels is a good trade off